### PR TITLE
esdm: 0.6.0 -> 1.0.0

### PR DESCRIPTION
**Pull request by thillux**
## Description of changes

ESDM 1.0.0 contains robustness and feature improvements. Besides support for accelerated crypto operations with Botan or OpenSSL, options which we used in practical deployments were added and/or made explicit.

Changelog: https://github.com/smuellerDD/esdm/blob/master/CHANGES.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thillux** [comment]
https://github.com/ofborg build pkgsCross.aarch64-multiplatform.esdm

**Comment by thillux** [comment]
Result of `nixpkgs-review pr 259945` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>esdm</li>
  </ul>
</details>

**Review by nikstur — APPROVED** [review]
This looks extremely complicated, but I guess the package is just so complicated.

In general it is a good idea to not reproduce too many options of the application in Nix. However, I'm not sure how much we could apply that here. Since you're the maintainer I think it's fine.

File: pkgs/os-specific/linux/esdm/default.nix [hunk 1]
```diff
@@ -9,67 +9,109 @@
...
 , linuxGetRandom ? true # enable linux getrandom support
-, esJitterRng ? true # enable support for the entropy source: jitter rng
+, hashSha512 ? false # set the conditioning hash: SHA2-512
+, hashSha3_512 ? true # set the conditioning hash: SHA3-512
+, openSSLRandProvider ? true # build ESDM provider for OpenSSL 3.x
+, botanRng ? true # build ESDM class for Botan 3.x
+
+# client-related options (handle with care, consult source code and meson options)
+# leave as is if in doubt
+, connectTimeoutExponent ? 28 # (1 << EXPONENT nanoseconds)
```
[the remark this answers is not in the document]
Comment (thillux) [reply pkgs/os-specific/linux/esdm/default.nix:36]: This is the default of the application, just exposed to be changed. I typically set 22, when I use ESDM, which boils down to a retry every 4ms.

**Review by ORichterSec — APPROVED** [review]

**Pull request merged**